components/esp32/cpu_start.c

```diff
@@ -88,12 +88,15 @@ extern volatile int port_xSchedulerRunning[2];
 
 static const char* TAG = "cpu_start";
 
+uint32_t g_boot_image_flash_addr;
+
 /*
  * We arrive here after the bootloader finished loading the program from flash. The hardware is mostly uninitialized,
  * and the app CPU is in reset. We do have a stack, so we can do the initialization in C.
+ *
+ * Boot loader passes flash offset of our image so we can later know for sure which partition was used for boot.
  */
-
-void IRAM_ATTR call_start_cpu0()
+void IRAM_ATTR call_start_cpu0(uint32_t boot_image_flash_addr)
```

Comment:
yeah, i actually looked at flash_mmap.c and couldn't figure it out quickly. this seemed more straightforward to me, and also more bullet-proof, because who knows what was subsequently done to mappings - we may be running from who knows where now (i can imagine user wishing to remap executable area somewhere else), what matters for OTA is what happened at boot. so i went for this approach.